cloudify_cli/cli/cfy.py

```diff
@@ -256,40 +246,77 @@ def parse_and_validate_label_to_delete(ctx, param, value):
         raise CloudifyValidationError(
             'ERROR: The `{0}` argument is empty'.format(param.name))
 
-    if ',' in value or value.count(':') > 1:
+    labels_list = get_formatted_labels_list(value, allow_only_key=True)
+    if len(labels_list) > 1:
         raise CloudifyValidationError(
             'LABEL can be either <key>:<value> or <key>')
+    [(label_key, label_value)] = labels_list[0].items()
+    if label_value:
+        return labels_list[0]
+    return label_key
 
-    label = value.split(':')
-    if len(label) == 1:
-        validate_param_value('The provided key', label[0])
-        return label[0]
-    else:
-        label_key, label_value = label
-        validate_param_value('The key of the provided label', label_key)
-        validate_param_value('The value of the provided label', label_value)
-        return {label_key: label_value}
 
+def get_formatted_labels_list(raw_labels_list, allow_only_key=False):
+    labels_list = []
+    if '\x00' in raw_labels_list:
+        raise CloudifyValidationError('Error: labels cannot contain null')
+    format_err_msg = 'Labels should be of the form <key>:<value>,<key>:<value>'
+    raw_labels_list = raw_labels_list.replace('\\,', '\x00').split(',')
```

Comment:
A general qualm: ASCII actually supports `\x1A` (Substitute character) for this. You can also use any of `\x1C` to `\x1F` and thus avoid the double substitution you had to make here

Reply:
Oh, I wonder what does the word `substitute means there. In some places I read, the `\x1A` is used to specify EOF. And I see the point of using the Control characters for this purpose, it's a good idea. I'll also disallow them.